Kittygram/ViewControllers/KittyDetails/KittyDetailsViewController.swift

```diff
@@ -7,18 +7,28 @@
 //
 
 import UIKit
+import RxSwift
+import RxCocoa
 
 class KittyDetailsViewController: UIViewController {
 
-    var kitty: Repository!
+    var kittyDetailsViewModel: KittyDetailsViewModel!
+    let disposeBag = DisposeBag()
     
     convenience init(kitty: Repository) {
         self.init()
-        
-        self.kitty = kitty
+        self.kittyDetailsViewModel = KittyDetailsViewModel(repository: kitty)
```

Comment:
Correct me if i wrong, is this your suggest is create KittyDetailsCordinator and KittyDetailsViewModel should be created in KittyDetailsCordinator?.

Could you please tell me the benefit  of that approach.